### DISCUSSION TITLE
Accessing buildId in the app

### DISCUSSION THREAD

**Discussion by trm217** [vercel/next.js#42738] [Help]
With `generateBuildId` we can specify a specific value for the buildId.
What I am missing however is a way to access the buildId on the client or server.

For example, for logging, it might be pretty useful if we could access the buildId from within the page. I suppose this isn't a surprising use case, so I wonder if the documentation is perhaps missing something in that regard.

**Comment by icyJoseph** [comment]
There was a couple of good recommendations here: https://github.com/vercel/next.js/discussions/36536 try some out and let me know which one works best for you :)

**Reply by trm217** [reply]
We use a workaround like this atm. but there should really be an API for it.

**Reply by icyJoseph** [reply]
In a way, there is `generateBuildId`, you can almost just invoke the same thing server side and send it as props to the client. I know a getBuildId would be awesome, but I am afraid that'd open up the door for various dangerous patterns.
